### PR TITLE
Fix compatibility with latest Github provider release

### PR DESCRIPTION
**Pull request by borisroman**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,3 +1,8 @@
 terraform {
-  required_version = ">= 0.12"
+  required_version = ">= 0.13"
+  required_providers {
```
Comment (svanharmelen) [inline versions.tf:3]: Please put a blank line in between these. And is this change TF 0.13 related? Guess this should still be able to be used by TF 0.12 as well rigth?

And I assume the GH provider also needs a minimum version?

File: main.tf [hunk 2]
```diff
@@ -12,6 +12,10 @@ locals {
   ])
 }
 
+data "github_repository" "default" {
+  name = var.name
+}
```
Comment (svanharmelen) [inline main.tf:17]: So how does this work for a new repo?

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: versions.tf (hunk 1 above)
Comment (borisroman) [reply versions.tf:4]: done & done

File: main.tf (hunk 2 above)
Comment (borisroman) [reply main.tf:17]: Check, fixed that as well.

File: main.tf [hunk 3]
```diff
@@ -12,6 +12,11 @@ locals {
   ])
 }
 
+data "github_repository" "default" {
+  count = var.create_repository ? 0 : 1
+  name  = var.name
```
Comment (svanharmelen) [inline main.tf:17]: Or:
```suggestion
  name  = var.create_repository ? github_branch_protection.default.name : var.name
```

And then you can omit the count here and use the data source ref in the ID loc

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**